legend-sdlc-generation-service/src/main/java/org/finos/legend/sdlc/generation/service/ServiceExecutionGenerator.java

```diff
@@ -173,6 +174,28 @@ private void writeJavaClass(GeneratedJavaCode generatedJavaClass) throws IOExcep
         }
     }
 
+    private void writeEnumJavaClass(GeneratedJavaCode generatedJavaClass) throws IOException
+    {
+        Path javaClassPath = this.javaSourceOutputDirectory.resolve(getJavaSourceFileRelativePath(generatedJavaClass.getClassName()));
+        byte[] content = generatedJavaClass.getText().getBytes(StandardCharsets.UTF_8);
+        if (Files.exists(javaClassPath))
+        {
+            byte[] foundContent = Files.readAllBytes(javaClassPath);
+            if (!Arrays.equals(content, foundContent))
+            {
+                throw new IOException("Duplicate file path found with different content for enum: " + javaClassPath);
+            }
+        }
+        else
+        {
+            Files.createDirectories(javaClassPath.getParent());
+            try (Writer writer = Files.newBufferedWriter(javaClassPath, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW))
+            {
+                writer.write(generatedJavaClass.getText());
+            }
```

Comment:
Since we've already converted the text to bytes, there's no need to do it again via a BufferedWriter. Just write the bytes.
```suggestion
            Files.write(javaClassPath, content, StandardOpenOption.CREATE_NEW);
```